deps/build.jl

```diff
@@ -1,28 +1,102 @@
 using CBindingGen
+using Libdl
+import Pkg.TOML
 import P4est_jll
 
-# Get include directory from jll package
-const incdir = joinpath(dirname(dirname(P4est_jll.libp4est_path)), "include")
+
+# setup configuration using ideas from MPI.jl
+config_toml = joinpath(first(DEPOT_PATH), "prefs", "P4est.toml")
+mkpath(dirname(config_toml))
+
+if !isfile(config_toml)
+  touch(config_toml)
+end
+
+config = TOML.parsefile(config_toml)
+
+# P4est.toml has 2 keys
+#  library  = "" (default) | library name/path
+#  include  = "" (default) | include name/path
+
+
+# Step 1: Check environment variables and update preferences accordingly
+if haskey(ENV, "JULIA_P4EST_LIBRARY")
+	config["library"] = ENV["JULIA_P4EST_LIBRARY"]
+else
+	config["library"] = ""
+end
+
+if haskey(ENV, "JULIA_P4EST_INCLUDE")
+	config["include"] = ENV["JULIA_P4EST_INCLUDE"]
+else
+	config["include"] = ""
+end
+
+
+open(config_toml, "w") do io
+	TOML.print(io, config)
+end
+
+
+
+# Step 2: Choose the library according to the settings
+if isempty(config["library"])
+	println("Use p4est library provided by P4est_jll")
+	p4est_library = P4est_jll.libp4est_path
+else
+	println("Use custom p4est library $(config["library"])")
+	p4est_library = config["library"]
+end
+
+
+
+# Step 3: Choose the include path according to the settings
+include_directories = String[]
+if isempty(config["include"])
+	println("Use p4est include path provided by P4est_jll")
+	push!(include_directories,
+				joinpath(dirname(dirname(P4est_jll.libp4est_path)), "include"))
+else
+	println("Use custom p4est include path $(config["include"])")
+	push!(include_directories, config["include"])
+end
+
+
+
+# Step 4: Generate binding using the include path according to the settings
 
 # Manually set header files to consider
-const hdrs = ["p4est.h", "p8est.h", "p4est_extended.h", "p8est_extended.h"]
+hdrs = ["p4est.h", "p4est_extended.h", "p8est.h", "p8est_extended.h"]
```

Comment:
What happened to p6est?

Reply:
You didn't have it in the version before on `master`, so I removed it here, too.

Reply:
Could you please add it again? I only did not have it in there because I've never used that part of the library wrapper efire, but others might find it useful.

Reply:
Done in 5b5099e